test/alternator/test_returnconsumedcapacity.py

```diff
@@ -80,3 +83,104 @@ def test_long_get(test_table):
+    assert 2 == response['ConsumedCapacity']["CapacityUnits"]
+
+    p = random_string()
+    c = random_string()
+    val = random_string()
+    val2 = random_string()
+    response = test_table.put_item(Item={'p': p, 'c': c, 'att': val, ('a' * KB): val2}, ReturnConsumedCapacity='TOTAL')
+    consumed_capacity = response['ConsumedCapacity']
+    assert 2 == consumed_capacity["CapacityUnits"]
+
+# This test validate that attribute names that are longer than one byte
+# (alternator tests default naming) still calculate WCU correctly.
+def test_long_put_varied_key(dynamodb):
+    with new_test_table(dynamodb,
+                        KeySchema=[ { 'AttributeName': 'p123', 'KeyType': 'HASH' },
+                    { 'AttributeName': 'c4567', 'KeyType': 'RANGE' }
+        ],
+        AttributeDefinitions=[
+                    { 'AttributeName': 'p123', 'AttributeType': 'S' },
+                    { 'AttributeName': 'c4567', 'AttributeType': 'S' },
+        ]) as table:
+        p = random_string()
+        c = random_string()
+        val = random_string()
+        combined_keys = "p123c4567attanother" # Takes all the keys and make one single string out of them
+        total_length = len(p) + len(c) + len(val) + len(combined_keys)
+
+        val2 = 'a' * (KB - total_length)  # val2 is a string that makes the total message length equals to 1KB
+        response = table.put_item(Item={'p123': p, 'c4567': c, 'att': val, 'another': val2}, ReturnConsumedCapacity='TOTAL')
+        assert 'ConsumedCapacity' in response
+        assert 1 == response['ConsumedCapacity']["CapacityUnits"]
+
+        response = table.put_item(Item={'p123': p, 'c4567': c, 'att': val, 'another': val2 + 'a'}, ReturnConsumedCapacity='TOTAL')  # Adding one byte, the total size is 1 byte more than 1KB
+        assert 2 == response['ConsumedCapacity']["CapacityUnits"]
+
+# this helper function mimic how DynamoDB calculate a number size
+# https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/CapacityUnitCalculations.html
+# `The size of a number is approximately (number of UTF-8-encoded bytes of attribute name) + (1 byte per two significant digits) + (1 byte)`
+def num_length(n):
+    s = str(n)
+    l = len(s)
+    if '-' in s:
+        l += 1
+    l += 1
+    l /= 2
+    l += 1
+    return int(l)
+
+# This test validate that numbers length are calculated correctly.
+# There are two difficulties, one, DynamoDB number calculation is approximate
+# and two, Alternator and DynamodDB uses slightly different encoding.
+# We assume that close proximity is ok.
+# The way the test work, it performs two put_item operations.
+# One that would be 5 bytes shorter than the approximation of 1KB
+# and one that is 5 bytes longer than that approximation.
+# this makes the test pass both on DynamoDB and Alternator
```

Comment:
Note that this way of testing is not perfect - it means you calculated the length of the integer "3" (for example) up to 5 bytes. Is this good enough approximation? I don't know.
But we can think about this later. I don't think it's really important that we have exactly the same formula as DynamoDB as long as we have a reasonable formula (and one day we'll need to document what it is).

Reply:
you are right, but we do test it with multiple magnitude numbers, so I assume it's close enough